experimental/devices/pca9685/pca9685.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package pca9685
+
+import (
+	"time"
+
+	"periph.io/x/periph/conn/gpio"
+	"periph.io/x/periph/conn/i2c"
+	"periph.io/x/periph/conn/physic"
+)
+
+// I2CAddr i2c default address.
+const I2CAddr uint16 = 0x40
+
+// PCA9685 Commands
+const (
+	mode1      byte = 0x00
+	mode2      byte = 0x01
+	subAdr1    byte = 0x02
+	subAdr2    byte = 0x03
+	subAdr3    byte = 0x04
+	prescale   byte = 0xFE
+	led0OnL    byte = 0x06
+	led0OnH    byte = 0x07
+	led0OffL   byte = 0x08
+	led0OffH   byte = 0x09
+	allLedOnL  byte = 0xFA
+	allLedOnH  byte = 0xFB
+	allLedOffL byte = 0xFC
+	allLedOffH byte = 0xFD
+
+	// Bits
+	restart byte = 0x80
+	sleep   byte = 0x10
+	allCall byte = 0x01
+	invrt   byte = 0x10
+	outDrv  byte = 0x04
```

Comment:
This comment is only FYI.

This device can sink 25mA and in totem-pole can source 10mA. You always put it in totem-pole at the moment.
I started adding `Drive()` to some GPIOs (like https://periph.io/x/periph/host/bcm283x#Pin.Drive) but I haven't defined an interface to gpio yet. I realize I should probably have both Sink and Source instead, so maybe I'll change that.